### PR TITLE
QA Process review

### PR DESCRIPTION
**Pull request by Miniplop**
This PR : 
- move some dependencies to devDependencies to avoid to install them in the user project
- add a try catch around the screenshoot controller to serve a proper error in case of internal errors
- fix a regression on the create slice modal where the wrong slice name format message disappear

### CODE REVIEW

File: packages/slice-machine/lib/models/common/Slice.ts [hunk 1]
```diff
@@ -21,6 +21,17 @@ const Slice = {
   },
 };
 
+export interface SliceCreateBody {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/lib/models/common/Slice.ts]: OPT: you know you could have a base type and extend it?
```
interface SliceBody {
  sliceName: string;
  from: string;
}

interface SliceCreateBody extends SliceBody {
   values?: { componentCode: string; model: Models.SliceAsObject };
}
```

File: packages/slice-machine/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "build-server": "tsc --project server/tsconfig.json",
     "build-next-app": "cross-env NODE_OPTIONS=--max_old_space_size=8192 next build",
     "export-build": "next export && npm run clean-build",
-    "bundle": "npm run build && npm run test && npm run export-build",
+    "bundle": "npm run test && npm run export-build",
```
Comment (MarcMcIntosh) [inline packages/slice-machine/package.json]: I think build is needed here?

File: packages/slice-machine/package.json [hunk 3]
```diff
@@ -59,14 +59,12 @@
     "esm": "^3.2.25",
     "express": "^4.17.1",
     "express-form-data": "^2.0.14",
-    "express-http-proxy": "^1.6.3",
```
Comment (MarcMcIntosh) [inline packages/slice-machine/package.json:62]: isn't this used server side?

File: packages/slice-machine/server/src/api/index.ts [hunk 4]
```diff
@@ -90,7 +74,7 @@ router.get(
   "/screenshot",
   // eslint-disable-next-line @typescript-eslint/no-misused-promises
   async function (
-    req: express.Request,
+    req: express.Request<undefined, undefined, undefined, ScreenshotRequest>,
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/src/api/index.ts]: what?

File: packages/slice-machine/server/src/api/index.ts [hunk 5]
```diff
@@ -107,12 +91,10 @@ router.post(
   "/custom-screenshot",
   // eslint-disable-next-line @typescript-eslint/no-misused-promises, @typescript-eslint/no-misused-promises, @typescript-eslint/no-explicit-any
   async function (req: any, res: express.Response): Promise<Express.Response> {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-member-access
+    /* eslint-disable */
     const payload = await customScreenshot(req.files.file, req.body);
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    if (payload.err) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/src/api/index.ts:113]: can this err?

File: packages/slice-machine/server/src/api/index.ts [hunk 6]
```diff
@@ -145,10 +127,6 @@ router.post(
   ): Promise<Express.Response> {
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
     const payload = await saveSlice(req);
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    if (payload.err) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/src/api/index.ts:149]: can this err?

File: packages/slice-machine/server/src/api/index.ts [hunk 7]
```diff
@@ -201,15 +178,11 @@ router.post(
...
   // eslint-disable-next-line @typescript-eslint/no-misused-promises
   async function (
-    req: express.Request,
+    req: express.Request<undefined, undefined, SaveCustomTypeBody>,
     res: express.Response
   ): Promise<Express.Response> {
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
     const payload = await saveCustomType(req);
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    if (payload.err) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/src/api/index.ts:210]: this can err?

File: packages/slice-machine/server/src/api/index.ts [hunk 8]
```diff
@@ -157,22 +135,21 @@ router.use(
...
     if (req.method === "POST") {
-      // eslint-disable-next-line @typescript-eslint/no-misused-promises, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
       payload = await createSlice(req.body);
     } else {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
       payload = await createSlice(req.query);
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
-    if (payload.err) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/src/api/index.ts:173]: can this err?

**Review by MarcMcIntosh — CHANGES_REQUESTED** [review]

File: packages/slice-machine/package.json (hunk 2 above)
Comment (Miniplop) [reply packages/slice-machine/package.json]: ouppsss yes of course :)

File: packages/slice-machine/package.json (hunk 3 above)
Comment (Miniplop) [reply packages/slice-machine/package.json:62]: not on prod, that's only for dev env

File: packages/slice-machine/server/src/api/index.ts (hunk 5 above)
Comment (Miniplop) [reply packages/slice-machine/server/src/api/index.ts:113]: nop the payload cannot contain the key err

File: packages/slice-machine/server/src/api/index.ts (hunk 6 above)
Comment (Miniplop) [reply packages/slice-machine/server/src/api/index.ts:149]: same here

File: packages/slice-machine/server/src/api/index.ts (hunk 7 above)
Comment (Miniplop) [reply packages/slice-machine/server/src/api/index.ts:210]: same :p

File: packages/slice-machine/server/src/api/index.ts (hunk 8 above)
Comment (Miniplop) [reply packages/slice-machine/server/src/api/index.ts:173]: re re re same :p

File: packages/slice-machine/lib/models/common/Slice.ts (hunk 1 above)
Comment (Miniplop) [reply packages/slice-machine/lib/models/common/Slice.ts]: Done

File: packages/slice-machine/server/src/api/index.ts (hunk 4 above)
Comment (Miniplop) [reply packages/slice-machine/server/src/api/index.ts]: Fixed

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**